torch/_subclasses/fake_tensor.py

```diff
@@ -1247,6 +1248,16 @@ def dispatch(self, func, types, args=(), kwargs=None):
             with self:
                 return func.prim_meta_impl(*args, **kwargs)
 
+        # Users can register FakeTensor rules for custom operators
+        # Call them if they exist.
+        if func.name() in torch._custom_op.global_registry:
```

Comment:
Any reason this table isn't indexed on func itself?

Reply:
Lifetime issues, but I am very close to declaring bankruptcy on lifetime issues anyways.

The problem is that once we create the OpOverload, if the operator gets unregistered and re-registered, invoking the OpOverload leads to segfaults. So as written in this PR, we index the registry on qualname to avoid creating the OpOverload.

This only really matters for testing, but I might (in a future PR) just develop a mechanism for testing to clear custom OpOverloads because testing needs to materialize the OpOverloads anyways when we do things like run make_fx.